common/src/main/java/org/corfudb/common/metrics/micrometer/MeterRegistryProvider.java

```diff
@@ -70,7 +72,8 @@ public static synchronized void init(Logger logger, Duration loggingInterval, St
 
         private static void init(Supplier<Optional<MeterRegistry>> meterRegistrySupplier) {
             if (meterRegistry.isPresent()) {
-                throw new IllegalStateException("Registry has already been initialized.");
+                log.warn("Registry has already been initialized.");
+                return;
```

Comment:
What if two different suppliers are passed? 

I don't think its a good idea to just silently print a warning message.

Reply:
I threw exception here and caught it at the places where we initialize a registry. We do need to ignore it for Corfu and Replication servers because otherwise, it would fail on restarts.